protostar/git.py

```diff
@@ -0,0 +1,215 @@
+
+    @staticmethod
+    def from_existing(path_to_repo: Path):
+        path = GitRepository.get_repo_root(path_to_repo)
+        return GitRepository(path)
+
+
+class GitRepository:
+    def __init__(self, path_to_repo: Path):
+        """
+        This class should not be instantiated directly, use `Git.init/clone/from_existing` instead.
+        """
+        self.path_to_repo = path_to_repo.resolve()
+
+    @staticmethod
+    def get_repo_root(path_to_repo: Path):
+        process = subprocess.run(
+            ["git", *CREDENTIALS, "rev-parse", "--show-toplevel"],
+            stdout=subprocess.PIPE,
+            stderr=subprocess.DEVNULL,
+            cwd=path_to_repo,
+            check=False,
+        )
+
+        if process.returncode:
+            raise NotARepositoryException(
+                f"{path_to_repo} is not a valid git repository."
+            )
+
+        path = Path(process.stdout.decode("utf-8").strip())
+        return path.resolve()
+
+    @wrap_git_exception
+    def init(self):
+        subprocess.run(
+            ["git", *CREDENTIALS, "init"],
+            **SHARED_KWARGS,
+            cwd=self.path_to_repo,
+        )
+
+    @wrap_git_exception
+    def clone(self, path_to_repo_to_clone: Path):
+        subprocess.run(
+            [
+                "git",
+                *CREDENTIALS,
+                "clone",
+                path_to_repo_to_clone,
+                self.path_to_repo.name,
+            ],
+            **SHARED_KWARGS,
+            cwd=self.path_to_repo.parent,
+        )
+
+    @wrap_git_exception
+    def add_submodule(
+        self,
+        url: str,
+        path_to_submodule: Path,
+        name: str,
+        branch: Optional[str] = None,
+        depth: int = 1,
+    ):
+        subprocess.run(
+            ["git", *CREDENTIALS, "submodule", "add"]
+            + (["-b", branch] if branch else [])  # (tag)
+            + [
+                "--name",
+                name,
+                "--depth",
+                str(depth),
+                url,
+                str(path_to_submodule.relative_to(self.path_to_repo)),
+            ],
+            **SHARED_KWARGS,
+            cwd=self.path_to_repo,
+        )
+
+    @wrap_git_exception
+    def update_submodule(self, path_to_submodule: Path, init=False):
+        subprocess.run(
+            ["git", *CREDENTIALS, "submodule", "update"]
+            + (["--init"] if init else [])
+            + [str(path_to_submodule)],
+            **SHARED_KWARGS,
+            cwd=self.path_to_repo,
+        )
+
+    @wrap_git_exception
+    def add(self, path_to_item: Path):
+        subprocess.run(
+            [
+                "git",
+                *CREDENTIALS,
+                "add",
+                str(path_to_item),
+            ],
+            **SHARED_KWARGS,
+            cwd=self.path_to_repo,
+        )
+
+    # pylint: disable=invalid-name
+    @wrap_git_exception
+    def rm(self, path_to_item: Path, force: bool = False):
```

Comment:
```suggestion
    def remove_submodule(self, submodule_name: str):
```
Just to be consistent with, `get`, `add`, and `update` submodule and to narrow down the use case.